src/transformers/trainer.py

```diff
@@ -553,7 +574,13 @@ def _get_train_sampler(self) -> Optional[torch.utils.data.sampler.Sampler]:
 
         else:
             if self.args.world_size <= 1:
-                return RandomSampler(self.train_dataset)
+                if version.parse(torch.__version__) < version.parse("1.6.0"):
+                    return RandomSampler(self.train_dataset)
+
+                # Torch generator were introduced in PyTorch 1.6.0.
+                generator = torch.Generator()
+                generator.manual_seed(int(torch.empty((), dtype=torch.int64).random_().item()))
+                return RandomSampler(self.train_dataset, generator=generator)
```

Comment:
```suggestion
                kwargs = {}
                if has_torch_generator():
                    generator = torch.Generator()
                    generator.manual_seed(int(torch.empty((), dtype=torch.int64).random_().item()))
                    kwargs.update(dict(generator=generator))
                return RandomSampler(self.train_dataset, **kwargs)
```
Here is another possible way (including the suggested earlier `has_torch_generator` which doesn't currently exist)